src/core/iomgr/pollset_posix.c

```diff
@@ -247,8 +246,11 @@ static void finish_shutdown(grpc_exec_ctx *exec_ctx, grpc_pollset *pollset) {
 }
 
 void grpc_pollset_work(grpc_exec_ctx *exec_ctx, grpc_pollset *pollset,
-                       grpc_pollset_worker *worker, gpr_timespec now,
+                       grpc_pollset_worker **worker_hdl, gpr_timespec now,
                        gpr_timespec deadline) {
+  grpc_pollset_worker worker;
+  *worker_hdl = &worker;
```

Comment:
That's about right.

In between we release the pollset mutex, meaning that calling code gets to see the 'published' value from other threads - and pass that value down to grpc_pollset_kick.